include/grpc++/impl/codegen/call.h

```diff
@@ -523,14 +525,30 @@ class CallOpClientRecvStatus {
 
 /// An abstract collection of CallOpSet's, to be used whenever
 /// CallOpSet objects must be thought of as a group. Each member
-/// of the group should have a shared_ptr back to the collection,
-/// as will the object that instantiates the collection, allowing
-/// for automatic ref-counting. In practice, any actual use should
-/// derive from this base class. This is specifically necessary if
-/// some of the CallOpSet's in the collection are "Sneaky" and don't
-/// report back to the C++ layer CQ operations
-class CallOpSetCollectionInterface
-    : public std::enable_shared_from_this<CallOpSetCollectionInterface> {};
+/// of the group should reference the collection, as will the object
+/// that instantiates the collection, allowing for ref-counting.
+/// Any actual use should derive from this base class. This is specifically
+/// necessary if some of the CallOpSet's in the collection are "Sneaky" and
+/// don't report back to the C++ layer CQ operations
+class CallOpSetCollectionInterface {
+ public:
+  CallOpSetCollectionInterface() {
+    gpr_atm_rel_store(&refs_, static_cast<gpr_atm>(1));
+  }
+  // always allocated against a call arena, no memory free required
+  static void operator delete(void* ptr, std::size_t size) {
```

Comment:
This probably doesn't work... but I suspect we can just hide the delete:

```
private:
  static void operator delete(void* ptr, size_t size);
```

... we'd need to change the delete then in ClientAsyncResponseReader from:
```
    if (collection_ != nullptr && collection_->Unref()) {
      delete collection_;
    }
```
to
```
    if (collection_ != nullptr && collection_->Unref()) {
      collection_->~CallOpSetCollection();
    }
```